From 290fb493e8d6412cd0a3d8b996aade6dc09ba7af Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:05:52 -0400
Subject: [tools] perf/report: Merge al->filtered with hist_entry->filtered

Message-id: <1410181752-25631-108-git-send-email-jolsa@redhat.com>
Patchwork-id: 91939
O-Subject: [PATCH RHEL7.1 BZ1134356 107/307] perf report: Merge al->filtered with hist_entry->filtered
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 2c86c7ca760634f09dcbd76069e5102b4de6f8f1
Author: Namhyung Kim <namhyung@kernel.org>
Date: Mon Mar 17 18:18:54 2014 -0300

description
===========
I.e. don't drop al->filtered entries, create the hist_entries and use
its ->filtered bitmap, that is kept with the same semantics for its
bitmap, leaving the filtering to be done at the hist_entry level, i.e.
in the UIs.

This will allow zooming in/out the filters.

[ yanked this out of a previous patch ]
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-report.c b/tools/perf/builtin-report.c
index c47bf58..a74059f 100644
--- a/tools/perf/builtin-report.c
+++ b/tools/perf/builtin-report.c
@@ -231,7 +231,7 @@ static int process_sample_event(struct perf_tool *tool,
   return -1;
  }
 
- if (al.filtered || (rep->hide_unresolved && al.sym == NULL))
+ if (rep->hide_unresolved && al.sym == NULL)
   return 0;
 
  if (rep->cpu_list && !test_bit(sample->cpu, rep->cpu_bitmap))
diff --git a/tools/perf/util/hist.c b/tools/perf/util/hist.c
index 9507f33..f38590d 100644
--- a/tools/perf/util/hist.c
+++ b/tools/perf/util/hist.c
@@ -422,7 +422,7 @@ struct hist_entry *__hists__add_entry(struct hists *hists,
    .weight = weight,
   },
   .parent = sym_parent,
-  .filtered = symbol__parent_filter(sym_parent),
+  .filtered = symbol__parent_filter(sym_parent) | al->filtered,
   .hists = hists,
   .branch_info = bi,
   .mem_info = mi,
-- 
1.7.1